src/internet_identity/src/main.rs

```diff
@@ -262,7 +322,7 @@ async fn add(user_number: UserNumber, device_data: DeviceData) {
             ));
         }
 
-        entries.push(device_data);
+        entries.push(DeviceDataInternal::from(device_data));
```

Comment:
Not sure if we should try to "fix" older devices for this user by setting explicitly to `Some` value for `purpose` and `key_type`. Thoughts?

Reply:
I probably wouldn't bother, given that we're already doing it lazily per account...